sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -5,183 +5,109 @@
+    withSpan<Options extends {
+        tracingOptions?: OperationTracingOptions;
+    }, Callback extends (updatedOptions: Options, span: Omit<TracingSpan, "end">) => ReturnType<Callback>>(name: string, operationOptions: Options, callback: Callback, spanOptions?: TracingSpanOptions): Promise<ReturnType<Callback>>;
 }
 
 // @public
-export function extractSpanContextFromTraceParentHeader(traceParentHeader: string): SpanContext | undefined;
-
-// @public
-export function getSpan(context: Context): Span | undefined;
-
-// @public
-export function getSpanContext(context: Context): SpanContext | undefined;
-
-// @public
-export function getTraceParentHeader(spanContext: SpanContext): string | undefined;
-
-// @public
-export function getTracer(): Tracer;
-
-// @public
-export function getTracer(name: string, version?: string): Tracer;
-
-// @public
-export type HrTime = [number, number];
-
-// @public
-export function isSpanContextValid(context: SpanContext): boolean;
-
-// @public
-export interface Link {
-    attributes?: SpanAttributes;
-    context: SpanContext;
+export interface TracingClientOptions {
+    namespace: string;
+    packageName: string;
+    packageVersion?: string;
 }
 
 // @public
-export interface OperationTracingOptions {
-    tracingContext?: Context;
+export interface TracingContext {
+    deleteValue(key: symbol): TracingContext;
+    getValue(key: symbol): unknown;
+    setValue(key: symbol, value: unknown): TracingContext;
 }
 
 // @public
-export function setSpan(context: Context, span: Span): Context;
-
-// @public
-export function setSpanContext(context: Context, spanContext: SpanContext): Context;
-
-// @public
-export interface Span {
-    addEvent(name: string, attributesOrStartTime?: SpanAttributes | TimeInput, startTime?: TimeInput): this;
-    end(endTime?: TimeInput): void;
+export interface TracingSpan {
+    end(): void;
     isRecording(): boolean;
-    recordException(exception: Exception, time?: TimeInput): void;
-    setAttribute(key: string, value: SpanAttributeValue): this;
-    setAttributes(attributes: SpanAttributes): this;
-    setStatus(status: SpanStatus): this;
-    spanContext(): SpanContext;
-    updateName(name: string): this;
+    recordException(exception: Error | string): void;
+    setAttribute(name: string, value: unknown): void;
+    setStatus(status: SpanStatus): void;
+    spanContext(): TracingSpanContext;
 }
 
 // @public
-export interface SpanAttributes {
-    [attributeKey: string]: SpanAttributeValue | undefined;
-}
-
-// @public
-export type SpanAttributeValue = string | number | boolean | Array<null | undefined | string> | Array<null | undefined | number> | Array<null | undefined | boolean>;
-
-// @public
-export interface SpanContext {
+export interface TracingSpanContext {
```

Comment:
Does core-tracing code ever need to touch the details of this? I would love to hide it as an opaque type if its only role is to get passed as part of TracingSpanLink

Actaully, can't we stick it in a `TracingContext` and have the instrumenter fish it back out? That would also let us have `parseTraceparentHeader` return the same that `createRequestHeaders` takes

Reply:
> Actaully, can't we stick it in a `TracingContext` and have the instrumenter fish it back out? That would also let us have `parseTraceparentHeader` return the same that `createRequestHeaders` takes

That's a great idea honestly! Let me play around with the idea and maybe get it into this PR if it's not too difficult

Reply:
Done!